File: 0108-convert-sorted-array-to-binary-search-tree/0108-convert-sorted-array-to-binary-search-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {number[]} nums
 * @return {TreeNode}
 */
var sortedArrayToBST = function(nums) {
    const makeBalancedBST = (nums) => {
        if(!nums.length) return null;
        const mid = Math.floor(nums.length / 2);
        const midNode = new TreeNode(nums[mid]);
        
        midNode.left = makeBalancedBST(nums.slice(0, mid));
        midNode.right = makeBalancedBST(nums.slice(mid + 1));

        return midNode
    }
    return makeBalancedBST(nums)
};

File: 0108-convert-sorted-array-to-binary-search-tree/0108-convert-sorted-array-to-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode sortedArrayToBST(int[] nums) {
        return makeBalancedBST(nums);
    }
    public TreeNode makeBalancedBST(int[] nums) {
        if(nums.length == 0) return null;
        int mid = nums.length / 2;
        TreeNode midNode = new TreeNode(nums[mid]);

        int[] leftHalf = Arrays.copyOfRange(nums, 0, mid);
        int[] rightHalf = Arrays.copyOfRange(nums, mid + 1, nums.length);

        midNode.left = makeBalancedBST(leftHalf);
        midNode.right = makeBalancedBST(rightHalf);

        return midNode;
    }
}